From 05b92eadd9c082a9b914c301437a77fccdca83c8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Jirman?= <megi@xff.cz>
Date: Mon, 15 Nov 2021 04:09:50 +0100
Subject: [PATCH 331/456] base: property: Swap order of search for connection
 to "devcon, graph"

This avoids confusing error in kernel log, when using devcon matching
in DT. Example:

"OF: graph: no port node found in /i2c@ff3d0000/typec-portc@22"

This suggest there's some error because graph based search failed,
but there is no error, because devcon based matching succeeds.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/base/property.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/base/property.c b/drivers/base/property.c
index e6497f6877ee..263e287b6e2c 100644
--- a/drivers/base/property.c
+++ b/drivers/base/property.c
@@ -1231,10 +1231,10 @@ void *fwnode_connection_find_match(struct fwnode_handle *fwnode,
 	if (!fwnode || !match)
 		return NULL;
 
-	ret = fwnode_graph_devcon_match(fwnode, con_id, data, match);
+	ret = fwnode_devcon_match(fwnode, con_id, data, match);
 	if (ret)
 		return ret;
 
-	return fwnode_devcon_match(fwnode, con_id, data, match);
+	return fwnode_graph_devcon_match(fwnode, con_id, data, match);
 }
 EXPORT_SYMBOL_GPL(fwnode_connection_find_match);
-- 
2.34.1

